featuretools/entityset/entity.py

```diff
@@ -120,6 +120,11 @@ def __init__(self, id, df, entityset, variable_types=None, name=None,
         if self.index is not None and self.index not in inferred_variable_types:
             self.add_variable(self.index, vtypes.Index)
 
+        # make sure index is at the beginning
+        index_variable = [v for v in self.variables
+                          if v.id == self.index][0]
+        self.variables = [index_variable] + [v for v in self.variables
```

Comment:
do we not care if the order of variable's doesn't match the underlying dataframe? I don't believe we assume that anywhere. can you think of any places?

Reply:
Why don't we reorder the columns in the dataframe as well?

Reply:
ya, let's do that